hindsight/io_modules/derived_stream/redshift/psv.lua

```diff
@@ -13,13 +13,13 @@ local string    = require "string"
 
 setfenv(1, M) -- Remove external access to contain everything in the module
 
-local esc_chars = { ["|"] = "\\|", ["\r"] = "\\r", ["\n"] = "\\n" }
+local esc_chars = { ["|"] = "\\|", ["\r"] = "\\r", ["\n"] = "\\n", ["\0"] = "0" }
```

Comment:
Yeah, the thing that sucks about the entire string discard check is that it requires yet another scan over every string but replacing the bad character with 0 \0 or "" isn't super useful and truncating at the first NUL just hides the problem.